src/rt/sched/schedulerthread.h

```diff
@@ -168,6 +187,129 @@ namespace verona::rt
       }
     }
 
+    /**
+     * Mute a set of cowns. This will add the cowns to the mute set of the
+     * mutor.
+     */
+    void mute(T** cowns, size_t count)
+    {
+      auto it = mute_map.find(mutor);
+      if (it == mute_map.end())
+      {
+        auto* set = ObjectMap<T*>::create(alloc);
+        it = mute_map.insert(alloc, std::make_pair(mutor, set)).second;
+      }
+      else
+      {
+        mutor->weak_release(alloc);
+      }
+      auto& mute_set = *it.value();
+
+      for (size_t i = 0; i < count; i++)
+      {
+        auto* cown = cowns[i];
+
+        if (
+          (state == ThreadState::PreScan) || (state == ThreadState::Scan) ||
+          (state == ThreadState::AllInScan))
+        { // Messages in this cowns queue must be scanned.
+          cown->schedule();
+          continue;
+        }
+
+        Systematic::cout() << "Mute " << cown << std::endl;
+        auto bp = cown->backpressure.load(std::memory_order_relaxed);
+        bp.mute();
+        cown->backpressure.store(bp, std::memory_order_release);
+        T::acquire(cown);
+        mute_set.insert(alloc, cown);
+      }
+
+      alloc->dealloc(cowns, count * sizeof(T*));
+    }
+
+    /**
+     * Scan the mute map for cowns that may be unmuted. If `force` is true, then
```

Comment:
Is this cowns that are unmuted but are still in the map, or cowns that are eligible to be unmuted?

Reply:
Only muted cowns are in the mute sets.

Reply:
the comment reads like there might be unmuted cowns in the mute map.